src/analyzer/protocol/ssh/ssh-analyzer.pac

```diff
@@ -211,12 +230,17 @@ refine flow SSH_Flow += {
 		%{
 		if ( ssh_server_host_key )
 			{
-			unsigned char digest[ZEEK_MD5_DIGEST_LENGTH];
-			zeek::detail::internal_md5(${key}.data(), ${key}.length(), digest);
+			unsigned char digest[ZEEK_SHA256_DIGEST_LENGTH];
+			auto ctx = zeek::detail::hash_init(zeek::detail::Hash_SHA256);
+			zeek::detail::hash_update(ctx, ${key}.data(), ${key}.length());
+			zeek::detail::hash_final(ctx, digest);
 
 			zeek::BifEvent::enqueue_ssh_server_host_key(connection()->zeek_analyzer(),
 				connection()->zeek_analyzer()->Conn(),
 				zeek::make_intrusive<zeek::StringVal>(fingerprint_md5(digest)));
```

Comment:
I'm a bit confused: Are we using the first few bytes of the SHA256 digest and format it in the `XX:XX...` MD5 format? I'd think that's not great :-)

Think this needs a completely separate `if ( ssh_server_host_key_fingerprint )`.

IMO, if it's possible, I don't think there's a reason to remove MD5 fingerprinting for those that want to continue using it (for historical reasons, ...). What do you think about deprecating it being included by default, but with an option to opt-in again, rather than deprecating with the idea of removing it.

Reply:
Similarly as for krb, but more work, a policy script to add back the MD5 fingerprints would be really nice, but I might also be overthinking this.